Entities/Player/Player.cs

```diff
@@ -97,14 +146,71 @@ private void HandleInput()
 			_velocity.Y = -_jumpVelocity;
 		}
 
-		if (Input.IsActionJustPressed("player_stealth_sprint"))
+		if (Input.IsActionPressed("player_action")
+		    && !GameManager.PlacementController.IsActive
+		    && _attackReady)
 		{
-			_isSprinting = !_isSprinting;
+			_attackReady = false;
+			Attack();
 		}
 	}
 
 	public bool ApplyDamage(DamageData damageInstance)
 	{
+		Health -= damageInstance.Damage;
+		_hurtSound.Play();
+		HealthChanged?.Invoke(Health);
+		if (Health <= 0)
+		{
+			Die(false);
+
+			return true;
+		}
+		
 		return false;
 	}
+
+	private void Attack()
+	{
+		if (_isSprinting)
+		{
+			AbilityController.ChangeAbility("player_stealth_sprint");	// TODO: This is gross. Destroy it.
+		}
+		
+		foreach (Node2D body in _damageArea.GetOverlappingBodies())
+		{
+			if (body is IDamageable damageable)
+			{
+				Vector2 direction = (body.GlobalPosition - GlobalPosition).Normalized();
+				var data = new DamageData
+				{
+					Damage = _damage,
+					Knockback = direction * _knockback,
+				};
+					
+				damageable.ApplyDamage(data);
+			}
+		}
+		
+		_attackParticle.Restart();
+		_attackAudioPlayer.Play();
+	}
+    
+	private void Flip(Vector2 direction)
+	{
+		var currentDirection = _damageArea.Position.Normalized();
+		var dot = currentDirection.Dot(direction);
+		if (currentDirection.Dot(direction) < 1)
```

Comment:
This process is technically incorrect, but it works fine in this circumstance. The check should be whether the dot product is positive or negative, not whether it is greater or less than 1. 

Which, because you're checking positive and negative, the normalization isn't necessary anymore.

The way this is written, if we decide the damage area should be moved up or down relative to the player, it will break.

Reply:
Actually kinda funny. The normalization technically is what causes this to break. If you don't normalize this, then it would always function as intended as long as the damage area x distance from the player is GREATER than 1. Kinda neat

Reply:
Fixed